libs/angular-integration-interface/src/lib/services/workspace.service.spec.ts

```diff
@@ -0,0 +1,454 @@
+    .subscribe((url) => {
+      expect(url).toBe('http://example.com/workspace/baseurl');
+      done();
+    });
+  });
+
+  //15
+  it('should find endpoint and return expected url despite of empty params"', (done) => {
+    service
+    .getUrl('onecx-workspace-ui', 'onecx-workspace', 'details', {})
+    .subscribe((url) => {
+      expect(url).toBe('http://example.com/workspace/baseurl');
+      done();
+    });
+  });
+
+  //16
+  it('should return well formed url with 2 params in endpoint', (done) => {
+    mockAppStateService.currentWorkspace$ = of({
+      baseUrl: 'http://example.com',
+      routes: [
+        {
+          appId: 'onecx-workspace-ui',
+          productName: 'onecx-workspace',
+          baseUrl: 'http://example.com/workspace/baseurl/',
+          endpoints: [
+            { name: 'details', path: '/details/{id}/{key}' },
+            { name: 'edit', path: '[[details]]' },
+            { name: 'change', path: '[[edit]]' },
+          ],
+        },
+      ],
+    });
+    
+    service.getUrl('onecx-workspace-ui', 'onecx-workspace', 'details', params)
+      .subscribe((url) => {
+        expect(url).toBe('http://example.com/workspace/baseurl/details/5/xy');
+        done();
+      })
+  })
+
+  //17
+  it('should return route.baseUrl when no endpoints are available"', (done) => {
+    mockAppStateService.currentWorkspace$ = of({
+      baseUrl: 'http://example.com',
+      routes: [
+        {
+          appId: 'onecx-workspace-ui',
+          productName: 'onecx-workspace',
+          baseUrl: 'http://example.com/workspace/baseurl',
+        },
+      ],
+    });
+    
+    service.getUrl('onecx-workspace-ui', 'onecx-workspace', 'detailswrong', params)
+      .subscribe((url) => {
+        expect(url).toBe('http://example.com/workspace/baseurl');
+        done();
+      })
+  })
+
+  //18
+  it('should return well formed url although double / are retrieved', (done) => {
+    mockAppStateService.currentWorkspace$ = of({
+      baseUrl: 'http://example.com',
+      routes: [
+        {
+          appId: 'onecx-workspace-ui',
+          productName: 'onecx-workspace',
+          baseUrl: 'http://example.com/workspace/baseurl/',
+          endpoints: [
+            { name: 'details', path: '//details/{id}' },
+            { name: 'edit', path: '[[details]]' },
+            { name: 'change', path: '[[edit]]' },
+          ],
+        },
+      ],
+    });
+    
+    service.getUrl('onecx-workspace-ui', 'onecx-workspace', 'details', params)
+      .subscribe((url) => {
+        expect(url).toBe('http://example.com/workspace/baseurl/details/5');
+        done();
+      })
+  })
+
+  it('it should return correct Endpoint with alias "', () => {
+    const endpoint = { name: 'edit', path: '[[details]]' }
+    const endpoints = [
+      { name: 'details', path: '/details/{id}' },
+      { name: 'edit', path: '[[details]]' },
+    ]
+
+    const resultEndpoint = service['dissolveEndpoint'](endpoint.name, endpoints)
```

Comment:
It is always better to only test the public interface of a service. If you can not find a way to test the code you are testing here with the public interface of the service it could be the case that the code is not needed at all